tests/test_georaster.py

```diff
@@ -1220,72 +1229,219 @@ def test_value_at_coords(self) -> None:
+
+        r.data.data[mask_pixel_artificially_set] = new_nodata
+        # We set the value as masked before unmasking to create the mask if it does not exist yet
+        r.data[mask_pixel_artificially_set] = np.ma.masked
+        r.data.mask[mask_pixel_artificially_set] = False
+
+        with pytest.warns(
+            UserWarning,
+            match=re.escape(
+                "New nodata value already found in the data array, the corresponding grid cells "
+                "will be indistinguishable from that updated from the old nodata value. Use "
+                "set_nodata(..., update_array=False) to avoid this behaviour."
+            ),
+        ):
+            r.set_nodata(nodata=new_nodata, update_mask=False)
+
+        # The nodata value should have been set in the metadata
+        assert r.nodata == new_nodata
+
+        # The array should have been updated
+        if old_nodata is not None:
+            index_old_nodata = r_copy.data.data == old_nodata
+            assert all(r.data.data[index_old_nodata] == new_nodata)
+        else:
+            index_old_nodata = np.zeros(np.shape(r.data.data), dtype=bool)
+
+        index_unchanged = np.logical_and(~index_old_nodata, ~mask_pixel_artificially_set)
+        # The rest of the array should be similarly unchanged
+        assert np.array_equal(r.data.data[index_unchanged], r_copy.data.data[index_unchanged])
+
+        # But the mask should still be the same
+        assert np.array_equal(np.ma.getmaskarray(r.data), np.ma.getmaskarray(r_copy.data))
+
+        # -- Fourth, test set_nodata() with both update_array=False and update_mask=False --
+        r = r_copy.copy()
+
+        r.set_nodata(nodata=new_nodata, update_array=False, update_mask=False)
+        r.data.data[mask_pixel_artificially_set] = new_nodata
+        # We set the value as masked before unmasking to create the mask if it does not exist yet
+        r.data[mask_pixel_artificially_set] = np.ma.masked
+        r.data.mask[mask_pixel_artificially_set] = False
+
+        # The nodata value should have been set in the metadata
+        assert r.nodata == new_nodata
+
+        # The array should not have been updated except for the pixel
+        assert np.array_equal(r.data.data[~mask_pixel_artificially_set], r_copy.data.data[~mask_pixel_artificially_set])
+        # And the mask neither
+        assert np.array_equal(np.ma.getmaskarray(r.data), np.ma.getmaskarray(r_copy.data))
+
+        # -- Fifth, let's check that errors are raised when they should --
+
+        # A ValueError if input nodata is neither a list, tuple, integer, floating
+        with pytest.raises(ValueError, match="Type of nodata not understood, must be list or float or int"):
+            r.set_nodata(nodata="this_should_not_work")  # type: ignore
+
+        # A ValueError if nodata value is incompatible with dtype
+        expected_message = r"nodata value .* incompatible with self.dtype .*"
+        if "int" in r.dtypes[0]:
```

Comment:
This will only be tested in one case (uint8). Maybe you could use the min value of the float128 dtype (which is very unlikely to be used in our sample data!).